src/ProductList.js

```diff
@@ -45,35 +48,55 @@ const filterByCategory = (setActiveFilters, activeFilters, id) => {
   }
 };
 
-const ProductList = ({ products, filterTypes }) => {
-  const [activeFilters, setActiveFilters] = useState([]);
+const ProductList = () => {
+  const [searchParams] = useSearchParams();
+  const categoryFilter = searchParams.get("category");
 
-  let filteredProducts = {
-    results:
-      activeFilters.length === 0
-        ? products.results
-        : products.results.filter((product) =>
-            activeFilters.includes(product.data.category.id)
-          ),
-  };
+  const [activeFilters, setActiveFilters] = useState(
+    categoryFilter !== null ? [categoryFilter] : []
+  );
```

Comment:
Esta línea yo la movería a una constante llamada `defaultFilter` y además la reduciría a lo siguiente.
```suggestion
    cosnt defaultFilter = categoryFilter ? [categoryFilter] : [];
    const [activeFilters, setActiveFilters] = useState( defaultFilter );
```